.circleci/config.yml

```diff
@@ -8,27 +8,27 @@ orbs:
 executors:
   build-executor:
     docker:
-      - image: circleci/rust:buster
+      - image: docker.io/libra/build_environment:latest
```

Comment:
How much do we gain in speed?

Reply:
not much 21 minutes with sccache, 29 w/o about the same.    Larger image size pulled for ever job.

Reply:
If we're going to go this far, then maybe its time to revive @metajack 's cargo-docker hack for caching builds of dependencies

Reply:
Cargo dep only shadow was a waste of time.  I dug deep in to it. Actually had it working when you were first on pto after I came on board.  mtimes were used by cargo to determine what to build, and at times code wasn't built when it needed to because the mtimes of the generated shadow were newer than what was pulled from git.

The right approach would be to pull all the code and build, and just pull on top of that.   Let cargo figure it out just as it would locally, bonus with sccache.

As of right now, I'm already eating tons of debt, and have little appetite for more.   Getting dev_setup.sh tested was my big push for this pr.   Might as well have new devs first experience not be a broken install script.

Maybe later, when it's planned work.